Java/core/src/main/java/com/amazon/randomcutforest/IComponentModel.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest;
+
+import com.amazon.randomcutforest.config.IDynamicConfig;
+import com.amazon.randomcutforest.executor.ITraversable;
+import com.amazon.randomcutforest.executor.IUpdatable;
+
+/**
+ *
+ * @param <PointReference> The internal point representation expected by the
+ *                         component models in this list.
+ * @param <Point>          The explicit data type of points being passed
+ */
+
+public interface IComponentModel<PointReference, Point>
+        extends ITraversable, IUpdatable<PointReference>, IDynamicConfig {
+}
```

Comment:
What is the etymology of "`IComponentModel`". I see here that it's something that's traversable, updatable, and configurable but seeing it used elsewhere it's not clear what is the purpose.

Also, why does it need `Point`? `IUpdatable` needs `PointReference` but I don't see anything to which the `Point` type is passed.

Reply:
Let me answer the easier one: the trees need Point because that is the explicit datatype exchanged with the pointstore. Now the interfaces on top of tree do not really need to know what Point is -- but there should be type check somewhere. 
For the harder question: it may make sense to break the pattern of one-tree-one-sampler down the road. Bootstrap sampling being the obvious use case. Other example can be grouped trees -- say 10 trees per sampler, and a few samplers. So the IComponent is a place where w can branch out.